src/AppBundle/Command/ImportSaleCommand.php

```diff
@@ -83,21 +81,13 @@ private function getSales($limit, $offset)
         return $response->json();
     }
 
-    private function saveSales($jsonSales, $existingInvoices)
+    private function saveSales($jsonSales, $repository)
     {
-        foreach ($jsonSales as $jsonSale) {
-            $exists = false;
-            foreach ($existingInvoices as $existing) {
-                if (
-                    $existing['invoice'] == $jsonSale['invoice'] &&
-                    $existing['licenseId'] == $jsonSale['licenseId'] &&
-                    $existing['pluginKey'] == $jsonSale['pluginKey']
-                ) {
-                    $exists = true;
-                }
-            }
+        $newCnt = 0;
 
-            if (!$exists) {
+        foreach ($jsonSales as $jsonSale) {
+            if ($repository->findIfSaleIsNew($jsonSale['invoice'], $jsonSale['licenseId'], $jsonSale['pluginKey'])) {
```

Comment:
Well, I thought it was a "cleaner" implementation and that it would improve performance because we aren't pulling a bunch of data out, but just letting sql check if the row exists. But if queries are relatively expensive, then I am probably wrong.